packages/webapp/src/containers/Finances/useTransactions.js

```diff
@@ -0,0 +1,241 @@
+    const labourItems = mapTasksToLabourItems(groupedTasks[date], taskTypes, users);
+    const amount = labourItems[LABOUR_ITEMS_GROUPING_OPTIONS.EMPLOYEE].reduce(
+      (sum, task) => sum + task.labourCost,
+      0,
+    );
+    if (amount > 0) {
+      groupedTransactions.push({
+        date,
+        transactionType: transactionTypeEnum.labourExpense,
+        amount: -amount,
+        items: labourItems,
+      });
+    }
+  });
+
+  return groupedTransactions;
+};
+
+const buildExpenseTransactions = ({ expenses, expenseTypes, dateFilter, expenseTypeFilter }) => {
+  return expenses
+    .filter(
+      (expense) =>
+        (!dateFilter ||
+          (moment(expense.expense_date).isSameOrAfter(dateFilter.startDate, 'day') &&
+            moment(expense.expense_date).isSameOrBefore(dateFilter.endDate, 'day'))) &&
+        expenseTypeFilter?.includes(expense.expense_type_id) &&
+        expense.value > 0,
+    )
+    .map((expense) => {
+      const expenseType = expenseTypes.find(
+        (expenseType) => expenseType.expense_type_id === expense.expense_type_id,
+      );
+      return {
+        icon: expenseType?.farm_id ? 'OTHER' : expenseType?.expense_translation_key,
+        date: expense.expense_date,
+        transactionType: transactionTypeEnum.expense,
+        typeLabel: expenseType?.farm_id
+          ? expenseType?.expense_name
+          : i18n.t(`expense:${expenseType?.expense_translation_key}`),
+        amount: -roundToTwoDecimal(expense.value),
+        note: expense.note,
+      };
+    });
+};
+
+const buildRevenueTransactions = ({
+  sales,
+  revenueTypes,
+  cropVarieties,
+  dateFilter,
+  revenueTypeFilter,
+}) => {
+  const filteredSales = sales.filter(
+    (sale) =>
+      (!dateFilter ||
+        (moment(sale.sale_date).isSameOrAfter(dateFilter.startDate, 'day') &&
+          moment(sale.sale_date).isSameOrBefore(dateFilter.endDate, 'day'))) &&
+      revenueTypeFilter?.includes(sale.revenue_type_id),
+  );
+
+  // We only want to show one revenue transaction per type per day. When expanding the item details on how that transaction was summed up from tasks will be displayed.
+  const groupedByRevenueTypeSales = groupBy(
+    filteredSales,
+    ({ revenue_type_id }) => revenue_type_id,
+  );
+  const groupedTransactions = [];
+
+  Object.keys(groupedByRevenueTypeSales).forEach((revenueTypeId) => {
+    const groupedByDateSales = groupBy(
+      groupedByRevenueTypeSales[revenueTypeId],
+      ({ sale_date }) => sale_date,
+    );
+    const revenueType = revenueTypes.find(
+      (revenueType) => revenueType.revenue_type_id == revenueTypeId,
+    );
+
+    Object.keys(groupedByDateSales).forEach((date) => {
+      const revenueItems = mapSalesToRevenueItems(
+        groupedByDateSales[date],
+        revenueTypes,
+        cropVarieties,
+      );
+      groupedTransactions.push({
+        icon: revenueType?.farm_id ? 'OTHER' : revenueType?.revenue_translation_key,
```

Comment:
This icon key is called `'CUSTOM'` in revenues.

This is lucky (or maybe @SayakaOno thought ahead and made it this way on purpose?!) because the revenue and expense icon keys won't conflict with each other when mixed in a single list.

Reply:
Ohhh good catch, will update!